src/OrleansRuntime/Core/InsideRuntimeClient.cs

```diff
@@ -336,6 +337,15 @@ public void SniffIncomingMessage(Message message)
                 try
                 {
                     var request = (InvokeMethodRequest) message.BodyObject;
+                    if (request.Arguments != null)
+                    {
+                        for (var i = 0; i < request.Arguments.Length; i++)
+                        {
+                            var arg = request.Arguments[i];
+                            if (!(arg is GrainCancellationToken)) continue;
+                            CancellationSourcesExtension.RegisterCancellationToken(target, request, i, logger);
+                        }
+                    }
```

Comment:
Can you move the entire looping of arguments inside of `RegisterCancellationToken` (and pluralize the method name)? It feels a little messy the way the responsibilities are split now, passing indices and so on.